### PR TITLE
DI-472 Add change to remove specified dates in the past

### PR DESCRIPTION
**Pull request by mabe13**
# Task Branch Pull Request

**<https://nhsd-jira.digital.nhs.uk/browse/DI-472>**

## Description of Changes

Removes past specified dates from incomming change events

## Type of change

Delete not appropriate

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Refactoring (non-breaking change which improves the structure of the code)

## Development Checklist

- [x] I have performed a self-review of my own code
- [x] I have run the [code formatting checks](../README.md#code-quality)
- [x] I have run the [code quality checks](../README.md#code-quality)
- [x] New code meets [standards](https://nhsd-confluence.digital.nhs.uk/display/DI/DI+Ways+of+Working) agreed by the team
- [x] Tests have added that prove my fix is effective or that my feature works (Integration tests)
- [x] I have updated Dependabot to include my changes (if applicable)

## Code Reviewer Checklist

- [x] I can confirm the changes have been tested or approved by a tester
- [x] I have checked any ignore commands for code linting tools and I agree that the code is safe

### CODE REVIEW

File: application/event_processor/changes.py [hunk 1]
```diff
@@ -54,6 +54,41 @@ def update_changes(changes: dict, change_key: str, dos_value: Any, nhs_uk_value:
...
+    nhs_spec_open_dates = nhs_entity.specified_opening_times
+    future_nhs_spec_open_dates = SpecifiedOpeningTime.remove_past_dates(nhs_entity.specified_opening_times)
+    if len(nhs_spec_open_dates) != len(future_nhs_spec_open_dates):
+        logger.debug(
+            "Removing Specified opening times that occur in the past",
+            extra={"all_nhs": nhs_spec_open_dates, "future_nhs": future_nhs_spec_open_dates},
+        )
+    compared = SpecifiedOpeningTime.equal_lists(dos_spec_open_dates, future_nhs_spec_open_dates)
+    if not compared:
+        logger.debug(
```
Comment (JackPlowman) [inline application/event_processor/changes.py]: Please can this log level be info

File: application/event_processor/changes.py [hunk 2]
```diff
@@ -54,6 +54,41 @@ def update_changes(changes: dict, change_key: str, dos_value: Any, nhs_uk_value:
...
+        )
+    return changes
+
+
+def __update_changes_with_std_opening_times(changes: dict, dos_service: DoSService, nhs_entity: NHSEntity) -> dict:
+    """Add week opening times to changes if they differ between dos_service and nhs_entity"""
+    dos_std_open_dates = dos_service.get_standard_opening_times()
+    nhs_std_open_dates = nhs_entity.standard_opening_times
+    if dos_std_open_dates != nhs_std_open_dates:
+        logger.debug(f"Standard weekly opening times not equal. dos={dos_std_open_dates} nhs={nhs_std_open_dates}")
```
Comment (JackPlowman) [inline application/event_processor/changes.py]: Please can this log level be info

File: application/event_processor/changes.py [hunk 3]
```diff
@@ -64,35 +99,15 @@ def update_changes_with_opening_times(changes: dict, dos_service: DoSService, nh
...
-            extra={"dos": dos_spec_open_dates, "nhs": nhs_spec_open_dates, "compared": compared},
-        )
-
-    # STANDARD OPENING TIMES (Comparing single StandardOpeningTimes Objects)
-    dos_std_open_dates = dos_service.get_standard_opening_times()
-    nhs_std_open_dates = nhs_entity.standard_opening_times
-    if dos_std_open_dates != nhs_std_open_dates:
-        logger.debug(f"Standard weekly opening times not equal. dos={dos_std_open_dates} nhs={nhs_std_open_dates}")
-        changes[OPENING_DAYS_KEY] = nhs_std_open_dates.export_cr_format()
+    changes = __update_changes_with_sp_opening_times(changes, dos_service, nhs_entity)
```
Comment (JackPlowman) [inline application/event_processor/changes.py]: Why is one `change = function` and one not?

**Review by JackPlowman — CHANGES_REQUESTED** [review]

**Review by JackPlowman — APPROVED** [review]
LGTM 👍

**Review by JackPlowman — DISMISSED** [review]
LGTM 👍

**Review by JackPlowman — APPROVED** [review]
LGTM 👍 , ready to merge when tested

**Review by Corbisc — APPROVED** [review]
This is functioning as expected in the 472 environment when tested.

**Review by JackPlowman — APPROVED** [review]

**Pull request merged**